src/v0/destinations/klaviyo/transformV2.js

```diff
@@ -0,0 +1,219 @@
+      category = CONFIG_CATEGORIES.TRACKV2;
+      response = trackOrScreenRequestHandler(message, category, destination);
+      break;
+    case EventType.GROUP:
+      category = CONFIG_CATEGORIES.GROUP;
+      response = groupRequestHandler(message, category, destination);
+      break;
+    default:
+      throw new InstrumentationError(`Event type ${messageType} is not supported`);
+  }
+  return response;
+};
+// {subscription:{}, event:{}, profile:{}}
+const processV2 = (event) => {
+  const response = processEvent(event);
+  const { destination } = event;
+  const respList = [];
+  if (response.profile) {
+    respList.push(buildRequest(response.profile, destination, CONFIG_CATEGORIES.IDENTIFYV2));
+  }
+  if (response.subscription) {
+    respList.push(
+      buildSubscriptionRequest(response.subscription, destination, CONFIG_CATEGORIES.TRACKV2),
+    );
+  }
+  if (response.event) {
+    respList.push(buildRequest(response.event, destination, CONFIG_CATEGORIES.TRACKV2));
+  }
+  return respList;
+};
+
+// This function separates subscribe, proifle and event responses from process () and other responses in chunks
+const getEventChunks = (input, subscribeRespList, profileRespList, eventRespList) => {
+  if (input.payload.subscription) {
+    subscribeRespList.push({ payload: input.payload.subscription, metadata: input.metadata });
+  }
+  if (input.payload.profile) {
+    profileRespList.push({ payload: input.payload.profile, metadata: input.metadata });
+  }
+  if (input.payload.event) {
+    eventRespList.push({ payload: input.payload.event, metadata: input.metadata });
+  }
+};
+
+const processRouterDestV2 = (inputs, reqMetadata) => {
+  const batchResponseList = [];
+  const batchErrorRespList = [];
+  const subscribeRespList = [];
+  const profileRespList = [];
+  const eventRespList = [];
+  const { destination } = inputs[0];
+  inputs.map((event) => {
+    try {
+      if (event.message.statusCode) {
+        // already transformed event
+        batchResponseList.push(fetchTransformedEvents(event));
+      } else {
+        // if not transformed
+        getEventChunks(
+          {
+            payload: processEvent(event),
+            metadata: event.metadata,
+          },
+          subscribeRespList,
+          profileRespList,
+          eventRespList,
+        );
+      }
+    } catch (error) {
+      const errRespEvent = handleRtTfSingleEventError(event, error, reqMetadata);
+      batchErrorRespList.push(errRespEvent);
+    }
+  });
+  const batchedResponseList = batchSubscriptionRequestV2(
+    subscribeRespList,
+    profileRespList,
+    destination,
+  );
+  const { anonymousTracking, identifiedTracking } = getTrackRequests(eventRespList, destination);
+
+  // We are doing to maintain event ordering basically once a user is identified klaviyo does not allow user tracking based upon anonymous_id only
+  batchResponseList.push(...anonymousTracking, ...batchedResponseList, ...identifiedTracking);
```

Comment:
Here ordering won't be applicable on basis of userId if we want that along with subscription batching then its not possible. Either we can support batching for subscription events or maintain the order.
Since ordering won't matter much as anonymous events will go first then identify and subscribe ones and then identified events so we didn't consider it much

Reply:
Thats fine but it might cause exceptions in server as server again validates by `userId` order. The solution we did for HS